ceno_emul/src/syscalls.rs

```diff
@@ -0,0 +1,63 @@
+use crate::{RegIdx, Tracer, VMState, Word, WordAddr, WriteOp};
+use anyhow::Result;
+
+pub mod keccak_permute;
+
+// Using the same function codes as sp1:
+// https://github.com/succinctlabs/sp1/blob/013c24ea2fa15a0e7ed94f7d11a7ada4baa39ab9/crates/core/executor/src/syscalls/code.rs
+
+pub const KECCAK_PERMUTE: u32 = 0x00_01_01_09;
+
+/// Trace the inputs and effects of a syscall.
+pub fn handle_syscall(vm: &VMState, function_code: u32) -> Result<SyscallEffects> {
+    match function_code {
+        KECCAK_PERMUTE => Ok(keccak_permute::keccak_permute(vm)),
+        _ => Err(anyhow::anyhow!("Unknown syscall: {}", function_code)),
```

Comment:
i think usually weak-typed anyhow err is not suitable for core lib functions, more suitable for app level errors.
We can add a TODO like "TODO: re-design err types when more precompile designs are done".

Reply:
Sure, I added a TODO. We can eventually revisit error-handling in this crate (https://github.com/scroll-tech/ceno/issues/391).